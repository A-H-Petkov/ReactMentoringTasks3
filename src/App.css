body {
  background-color: #282c34;
}

.App {
  /* text-align: center; */
  margin: 0;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

.App-header,
main {
  background-color: transparent;
  min-height: 50vh;
  padding: 40px 60px;
  position: relative;
  /* display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white; */
}

.app-logo {
  width: 140px;
  float: left;
  background-color: rgba(0, 0, 0, 0.14);
  border-radius: 8px;
}

.add-movie-btn {
  float: right;
}

.search-panel {
  clear: both;
  padding: 30px;
}

.search-panel-title {
  color: #61dafb;
  font-size: 28px;
  font-weight: 200;
  margin-bottom: 24px;
}

.posters-panel {
  background-image: url(./resources/Movie_Posters.jpg);
  
  /*Add blur effect*/
  filter: blur(8px);
  -webkit-filter: blur(8px);

  /* Full height */
  height: 100%;

  /* Center and scale the image nicely */
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;

  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: -1;
}

.catalog-controls {
  display: flex;
  justify-content: space-between;
  margin-bottom: 20px;
}

.sorting-label {
  color: antiquewhite;
  font-size: 18px;
  line-height: 38px;
  margin-right: 20px;
}

.catalog-container {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  align-items: flex-end;
  margin: 0 -18px;
}

.catalog-item {
  margin: 16px;
  flex: 1 1 30%;
  position: relative;
}

.movie-title-row h6, 
.movie-title-row span {
  display: inline-block;
}

.movie-title-row span {
  float: right;
  padding: 2px 8px;
  border: 1px solid gray;
  border-radius: 4px;
}

.app-footer {
  position: relative;
  padding: 24px;
}

.app-footer-logo {
  width: 160px;
  display: block;
  margin: 0 auto;
  background-color: rgba(0, 0, 0, 0.14);
  border-radius: 8px;
}

.item-dot-menu {
  position: absolute;
  top: 20px;
  right: 20px;
  height: 40px;
  width: 40px;
  background-color: rgba(0, 0, 0, 0.5);
  text-align: center;
  border-radius: 50%;
  color: whitesmoke;
  font-size: 38px;
  line-height: 16px;
  -webkit-transform:rotate(90deg);
  -moz-transform:rotate(90deg);
  -o-transform: rotate(90deg);
  -ms-transform:rotate(90deg);
  transform: rotate(90deg);
  display: none;
}

.catalog-item:hover .item-dot-menu {
  display: block;
}