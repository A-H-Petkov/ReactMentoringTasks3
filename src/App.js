import React, { useState } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import Modal from 'react-modal';
import ErrorBoundary from './ErrorBoundary';
import Catalog from './components/Catalog/Catalog';
import Header from './components/Header';
import DetailsPanel from './components/DetailsPanel'
import Footer from './components/Footer';
import ModalContent from './components/ModalContent'
import { 
  filterOptions, 
  sortingOptions, 
  defaultMovieList,
  modalTypes,
  modalStyles, 
} from './mockData/data';
import {
  updateMovieList
} from './helpers/listHelpers';
import './App.css';



Modal.setAppElement('#root');

const App = () => {

  const [openModal, setOpenModal] = useState('');
  const [movieList, setMovieList] = useState(defaultMovieList);
  const [stagedMovie, setStagedMovie] = useState(null);
  const [detailedPreview, setDetailedPreview] = useState(null);

  const prepareModalData = (setType, movie) => {
    setOpenModal(setType); 
    setStagedMovie(movie);
  };

  const closeModal = () => { 
    setOpenModal(''); 
    setStagedMovie(null);
  };

  const closeDetails = () => { 
    setStagedMovie(null)
  }

  const confirmModal = (data = {}) => {
    const updatedList = updateMovieList(movieList, openModal, stagedMovie, data);
    setMovieList(updatedList);
    closeModal();
  }

  const openDetailed = (item) => {
    setDetailedPreview(item)
  }

  return (
    <ErrorBoundary>
      <div className="App">
        {
          detailedPreview !== null 
          ?
          <DetailsPanel
            movie={detailedPreview}
            closeDetails={closeDetails}
          />   
          :
          <Header
          setOpenModal={prepareModalData} 
          modalTypes={modalTypes}
        />
        }
        
        <Catalog 
          filterOptions={filterOptions}
          sortingOptions={sortingOptions}
          movieList={movieList}
          setOpenModal={prepareModalData}
          modalTypes={modalTypes}
          openDetailed={openDetailed}
        />      
        <Footer />
      </div>
      <Modal 
        isOpen={openModal !== ''} 
        style={modalStyles}
        onRequestClose={closeModal}
      >
        <ModalContent
          closeModal={closeModal} 
          modalTypes={modalTypes}
          activeType={openModal}
          confirmModal={confirmModal}
          stagedMovie={stagedMovie}
        />
      </Modal>
    </ErrorBoundary>
  );
}

export default App;
