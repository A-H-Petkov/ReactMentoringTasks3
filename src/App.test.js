import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
// import Modal from 'react-modal';

test('renders learn react link', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
}); 
