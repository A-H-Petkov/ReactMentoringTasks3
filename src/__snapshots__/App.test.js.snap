// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="My_movies"
        class="app-logo"
        src="site-logo.png"
      />
      <button
        class="add-movie-btn btn btn-primary"
        type="button"
      >
        + ADD MOVIE
      </button>
      <div
        class="search-panel"
      >
        <h3
          class="search-panel-title"
        >
          FIND YOUR MOVIE
        </h3>
        <div
          class="mb-3 input-group"
        >
          <input
            aria-describedby="basic-addon2"
            aria-label="What do you want to watch?"
            class="form-control"
            placeholder="What do you want to watch?"
          />
          <div
            class="input-group-append"
          >
            <button
              class="btn btn-primary"
              id="basic-addon2"
              type="button"
            >
              SEARCH
            </button>
          </div>
        </div>
      </div>
      <div
        class="posters-panel"
      />
    </header>
    <main>
      <div
        class="catalog-controls"
      >
        <div
          aria-label="Basic example"
          class="btn-group btn-group-toggle"
          role="group"
        >
          <label
            class="btn active btn-outline-primary"
          >
            <input
              autocomplete="off"
              checked=""
              name="radio"
              type="radio"
              value="1"
            />
            ALL
          </label>
          <label
            class="btn btn-outline-primary"
          >
            <input
              autocomplete="off"
              name="radio"
              type="radio"
              value="2"
            />
            COMEDY
          </label>
          <label
            class="btn btn-outline-primary"
          >
            <input
              autocomplete="off"
              name="radio"
              type="radio"
              value="3"
            />
            DRAMA
          </label>
          <label
            class="btn btn-outline-primary"
          >
            <input
              autocomplete="off"
              name="radio"
              type="radio"
              value="4"
            />
            CRIME
          </label>
          <label
            class="btn btn-outline-primary"
          >
            <input
              autocomplete="off"
              name="radio"
              type="radio"
              value="5"
            />
            HORROR
          </label>
          <label
            class="btn btn-outline-primary"
          >
            <input
              autocomplete="off"
              name="radio"
              type="radio"
              value="6"
            />
            ACTION
          </label>
          <label
            class="btn btn-outline-primary"
          >
            <input
              autocomplete="off"
              name="radio"
              type="radio"
              value="7"
            />
            SCI-FI
          </label>
          <label
            class="btn btn-outline-primary"
          >
            <input
              autocomplete="off"
              name="radio"
              type="radio"
              value="8"
            />
            DOCUMENTARY
          </label>
        </div>
        <div
          class="btn-group"
          role="group"
        >
          <span
            class="sorting-label"
          >
            SORT BY: 
          </span>
          <div
            class="dropdown"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              class="dropdown-toggle btn btn-primary"
              id="dropdown-basic-button"
              type="button"
            >
              RELEASE DATE
            </button>
          </div>
        </div>
      </div>
      <div>
        <span
          class="sorting-label"
        >
          <b>
            6
          </b>
           MOVIES FOUND
        </span>
      </div>
      <div
        class="catalog-container"
      >
        <figure
          class="catalog-item figure"
        >
          <img
            alt="171x180"
            class="figure-img img-fluid"
            src="https://cdn.collider.com/wp-content/uploads/2020/03/black-widow-poster.jpg"
          />
          <figcaption
            class="figure-caption"
          >
            <div
              class="movie-title-row"
            >
              <h6>
                Black widow
              </h6>
              <span>
                2020
              </span>
            </div>
            <div>
              ACTION
            </div>
          </figcaption>
          <div
            class="item-dot-menu dropdown btn-group"
            role="group"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              class="item-dot-menu-trigger dropdown-toggle btn btn-primary"
              type="button"
            >
              ⋮
            </button>
          </div>
        </figure>
        <figure
          class="catalog-item figure"
        >
          <img
            alt="171x180"
            class="figure-img img-fluid"
            src="https://i.pinimg.com/originals/c6/d4/e2/c6d4e2803a467b99ede238fee57441a7.jpg"
          />
          <figcaption
            class="figure-caption"
          >
            <div
              class="movie-title-row"
            >
              <h6>
                Interstellar
              </h6>
              <span>
                2014
              </span>
            </div>
            <div>
              ACTION, DRAMA, SCI-FI
            </div>
          </figcaption>
          <div
            class="item-dot-menu dropdown btn-group"
            role="group"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              class="item-dot-menu-trigger dropdown-toggle btn btn-primary"
              type="button"
            >
              ⋮
            </button>
          </div>
        </figure>
        <figure
          class="catalog-item figure"
        >
          <img
            alt="171x180"
            class="figure-img img-fluid"
            src="https://cdn.shopify.com/s/files/1/1416/8662/products/mad_max_fury_road_2015_advance_original_film_artB_69310cd2-a499-45fc-a12d-df89480c4c99_2000x.jpg?v=1573593327"
          />
          <figcaption
            class="figure-caption"
          >
            <div
              class="movie-title-row"
            >
              <h6>
                MadMax
              </h6>
              <span>
                2015
              </span>
            </div>
            <div>
              ACTION
            </div>
          </figcaption>
          <div
            class="item-dot-menu dropdown btn-group"
            role="group"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              class="item-dot-menu-trigger dropdown-toggle btn btn-primary"
              type="button"
            >
              ⋮
            </button>
          </div>
        </figure>
        <figure
          class="catalog-item figure"
        >
          <img
            alt="171x180"
            class="figure-img img-fluid"
            src="https://cdn.shopify.com/s/files/1/1416/8662/products/blade_runner_2049_2017_advance_original_film_art_9eeddf62-08a7-4bb2-9e09-9f21f318c14c_2000x.jpg?v=1578262145"
          />
          <figcaption
            class="figure-caption"
          >
            <div
              class="movie-title-row"
            >
              <h6>
                Blade runner 2049
              </h6>
              <span>
                2017
              </span>
            </div>
            <div>
              ACTION, SCI-FI
            </div>
          </figcaption>
          <div
            class="item-dot-menu dropdown btn-group"
            role="group"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              class="item-dot-menu-trigger dropdown-toggle btn btn-primary"
              type="button"
            >
              ⋮
            </button>
          </div>
        </figure>
        <figure
          class="catalog-item figure"
        >
          <img
            alt="171x180"
            class="figure-img img-fluid"
            src="https://image.tmdb.org/t/p/original/ohS07RD5jpfUjEkYBC4q6MKaA6E.jpg"
          />
          <figcaption
            class="figure-caption"
          >
            <div
              class="movie-title-row"
            >
              <h6>
                Doctor Sleep
              </h6>
              <span>
                2019
              </span>
            </div>
            <div>
              ACTION, HORROR
            </div>
          </figcaption>
          <div
            class="item-dot-menu dropdown btn-group"
            role="group"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              class="item-dot-menu-trigger dropdown-toggle btn btn-primary"
              type="button"
            >
              ⋮
            </button>
          </div>
        </figure>
        <figure
          class="catalog-item figure"
        >
          <img
            alt="171x180"
            class="figure-img img-fluid"
            src="https://cdn.collider.com/wp-content/uploads/2014/02/jodorowskys-dune-poster.jpg"
          />
          <figcaption
            class="figure-caption"
          >
            <div
              class="movie-title-row"
            >
              <h6>
                DUNE
              </h6>
              <span>
                2014
              </span>
            </div>
            <div>
              DOCUMENTARY
            </div>
          </figcaption>
          <div
            class="item-dot-menu dropdown btn-group"
            role="group"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              class="item-dot-menu-trigger dropdown-toggle btn btn-primary"
              type="button"
            >
              ⋮
            </button>
          </div>
        </figure>
      </div>
    </main>
    <footer
      class="app-footer"
    >
      <img
        alt="My_movies"
        class="app-footer-logo"
        src="site-logo.png"
      />
      <div
        class="posters-panel"
      />
    </footer>
  </div>
</DocumentFragment>
`;
